### PR TITLE
627 - Set IBM Notes credentials from admin interface.

### PR DESCRIPTION
**Pull request by amiedes**
Connects to #627 

### What does this PR do?

This PR adds the possibility for a site admin to set IBM Notes credentials from a GUI.

Also, credentials are now stored as `GobiertoModuleSettings` instead of
`GobiertoPeople::Setting`. This commit contains a migration to perform
the corresponding data transformations.

Also the exception catching / error handling for edge cases where the API server may be down, no events are returned or the provided credentials are wrong.

### How should this be manually tested?

IBM Notes integration should keep working after this update.

### Does this PR changes any configuration file?

No

### CODE REVIEW

**Comment by ferblape** [comment]
Why is this branch not in master (and deployed to staging) and why is a WIP? Everything that is required to be reviewed should be in master, and everything that is a WIP should be explicit about why is WIP.

File: app/views/gobierto_admin/gobierto_people/configuration/settings/edit.html.erb [hunk 1]
```diff
@@ -39,6 +39,17 @@
...
       </div>
+
+      <div class="form_item input_text">
+        <%= f.label :ibm_notes_usr, t(".ibm_notes_usr") %>
+        <%= f.text_field :ibm_notes_usr, placeholder: t(".placeholders.ibm_notes_usr") %>
+      </div>
+
+      <div class="form_item input_text">
+        <%= f.label :ibm_notes_pwd, t(".ibm_notes_pwd") %>
+        <%= f.text_field :ibm_notes_pwd, placeholder: t(".placeholders.ibm_notes_pwd") %>
```
Comment (ferblape) [inline app/views/gobierto_admin/gobierto_people/configuration/settings/edit.html.erb]: This should be a password_field, to be more private.

File: app/forms/gobierto_admin/gobierto_people/settings_form.rb [hunk 2]
```diff
@@ -65,6 +75,14 @@ def save_settings
           settings_attributes.home_text_en = home_text_en
           settings_attributes.submodules_enabled = submodules_enabled.select{|m| m.present?}
           settings_attributes.calendar_integration = calendar_integration
+
```
Comment (ferblape) [inline app/forms/gobierto_admin/gobierto_people/settings_form.rb]: I think we should add a validation in the form: if the user selects IBM Notes integration but user and password are empty, we should display an error message that credentials are required.

File: app/views/gobierto_admin/gobierto_people/configuration/settings/edit.html.erb [hunk 3]
```diff
@@ -39,6 +39,17 @@
...
+      </div>
+
```
Comment (ferblape) [inline app/views/gobierto_admin/gobierto_people/configuration/settings/edit.html.erb]: Are you going to add the javascript code to clean the form? Is that why it's a WIP?

**Review by ferblape — CHANGES_REQUESTED** [review]
Code looks good. Please, add the changes I have requested and feel free to merge in master again and release to staging.

**Comment by amiedes** [comment]
I marked this as WIP because wanted to add also the JS code and the solution for the issues exposed in #627. I didn't deploy to staging because I wanted you to take a look at the migration script previously.

I'll try to be explicit about this things from now on 👌

**Comment by amiedes** [comment]
Deployed to staging 🚀 

The migration worked correctly but since it only deleted "deprecated" `GobiertoPeople::Setting` containing the key `IBM_NOTES_etc.`, I had to delete old `LOTUS_NOTES_etc.` settings manually from the console.

A few notes:

* I think the feedback provided by the error messages shown in the interface can be improved. Right now it only says "An error occured when saving this settings", but I couldn't guess how to customize it to say "Credentials can't be blank" etc.

**Review by ferblape — APPROVED** [review]

**Pull request merged**